devito/operator/operator.py

```diff
@@ -495,7 +495,11 @@ def _prepare_arguments(self, autotune=None, **kwargs):
                 # E.g., SubFunctions
                 continue
             for k, v in p._arg_values(**kwargs).items():
-                if k in args and args[k] != v:
+                try:
+                    val_test = k in args and (args[k] != v).all()
```

Comment:
the .all here is potentially expensive if a big numpy array? there's something weird going on here